### DISCUSSION TITLE
Suggestion: Replace Ngrok with Expose

### DISCUSSION THREAD

**Discussion by mathmul** [leokhoa/laragon#975] [Ideas]
### Intro (optional)

@leokhoa I've been using Share option in Laragon since I first downloaded it, and initially it blew my mind (Laragon itself and) that I can share local code to whomever I want. After a while I found some limitations of it. I avoid WordPress sites if I can help it, but every now and then... bleh.

I found that I quickly exceeded the limit of connections and the tunnel was broken. Went to Ngrok website, downloaded their stuff and authenticated my newly made account. Issue after issue (20min exercise), and I decided to look for alternatives.

What I found blew my mind the same way as my first usage of Laragon.

# THE POINT

One of THE masters of development world, Marcel Pociot, made a pure PHP (composer installed) package [Expose](https://github.com/beyondcode/expose). It's a-ducking-mazing! [Easiest installation](https://beyondco.de/docs/expose/getting-started/installation) ever, and it just works.

### Outro (optional)

I don't really care if this suggestion for replacement will be accepted or not, since I will definitely be using Expose in the future, but just in case you don't know about, here's a heads-up before the v5 release.

Though it might not even be possible to use, since it's PHP only. I don't know anything about <strike>your</strike> Laragon's internals.

Huge fan <3

PS: Sorry to everybody else who is, like me, eagerly waiting for the new release, if this might prolong the release date even further. Good things are worth waiting for!

**Comment by ghost** [comment]
I would love to see an ngrok alternative - I have experienced the same issues!

**Comment by mucahidbaris** [comment]
+1

**Comment by Gentleman03** [comment]
+999, that's the only thing I don't like about laragon. Ngrok is such a pain in :)

**Comment by mathmul** [comment]
@leokhoa Does this mean it's being taken into consideration now?
